Numismatic References and Citations

eliotjordan opened the pull request:
Closes #2115

hackmastera commented on app/controllers/numismatic_citations_controller.rb:
```diff
@@ -0,0 +1,46 @@
+  before_action :load_numismatic_references, only: [:new, :edit]
+
+  def create
+    # Save append_id and then remove from resource params
```
can you add why to this comment?

hackmastera replied:
you're using the append_id list to get the parents to attach to?

eliotjordan replied:
Yes. The append_id param is used to retrieve the parent numismatic issue or coin. The citation id is then inserted into the parent `numismatic_citation_ids` array.

eliotjordan commented:
@escowles I fixed an issue with one of the new model comments and expanded on the controller comment a little.

eliotjordan commented:
@hackmastera This is ready for another review.

- Updated to not overload the `append_id` parameter and moved logic to the change set persister.
- Rebased against recent Figgy file structure re-org.
